setup.py

```diff
@@ -1,107 +1 @@
-"""
-import os
-import setuptools
-import sys
-
-from jupyter_packaging import (
-    create_cmdclass,
-    install_npm,
-    ensure_targets,
-    combine_commands,
-)
-
-# ensure the current directory is on sys.path
-# so versioneer can be imported when pip uses
-# PEP 517/518 build rules.
-# https://github.com/python-versioneer/python-versioneer/issues/193
-sys.path.append(os.path.dirname(__file__))
-
-import versioneer
-
-HERE = os.path.abspath(os.path.dirname(__file__))
-
-# The name of the project
-name = "dask_labextension"
-
-# Get our version
-version = versioneer.get_version()
-
-lab_path = os.path.join(HERE, name, "labextension")
-
-# Representative files that should exist after a successful build
-jstargets = [
-    os.path.join(HERE, "lib", "index.js"),
-    os.path.join(lab_path, "package.json"),
-]
-
-package_data_spec = {name: ["*"]}
-
-labext_name = "dask-labextension"
-
-data_files_spec = [
-    ("share/jupyter/labextensions/%s" % labext_name, lab_path, "**"),
-    ("share/jupyter/labextensions/%s" % labext_name, HERE, "install.json"),
-    (
-        "etc/jupyter/jupyter_server_config.d",
-        "jupyter-config/jupyter_server_config.d",
-        "dask_labextension.json",
-    ),
-    (
-        "etc/jupyter/jupyter_notebook_config.d",
-        "jupyter-config/jupyter_notebook_config.d",
-        "dask_labextension.json",
-    ),
-]
-
-cmdclass = versioneer.get_cmdclass(
-    create_cmdclass(
-        "jsdeps", package_data_spec=package_data_spec, data_files_spec=data_files_spec
-    )
-)
-
-cmdclass["jsdeps"] = combine_commands(
-    install_npm(HERE, build_cmd="build:prod", npm=["jlpm"]),
-    ensure_targets(jstargets),
-)
-
-with open("README.md", "r") as fh:
-    long_description = fh.read()
-
-setup_args = dict(
-    name=name,
-    version=version,
-    url="https://github.com/dask/dask-labextension",
-    author="Ian Rose, Matt Rocklin, Jacob Tomlinson",
-    description="A JupyterLab extension for Dask.",
-    long_description=long_description,
-    long_description_content_type="text/markdown",
-    cmdclass=cmdclass,
-    packages=setuptools.find_packages(),
-    install_requires=[
-        "bokeh >=1.0.0,!=2.0.0",
-        "distributed>=1.24.1",
-        "jupyter-server-proxy>=1.3.2",
-        "jupyterlab>=3.0.0",
-    ],
-    zip_safe=False,
-    include_package_data=True,
-    python_requires=">=3.6",
-    license="BSD-3-Clause",
-    platforms="Linux, Mac OS X, Windows",
-    keywords=["dask", "Jupyter", "JupyterLab", "JupyterLab3"],
-    classifiers=[
-        "License :: OSI Approved :: BSD License",
-        "Programming Language :: Python",
-        "Programming Language :: Python :: 3",
-        "Programming Language :: Python :: 3.6",
-        "Programming Language :: Python :: 3.7",
-        "Programming Language :: Python :: 3.8",
-        "Framework :: Jupyter",
-    ],
-)
-
-
-if __name__ == "__main__":
-    setuptools.setup(**setup_args)
+# this file intentionally left blank for legacy tools to find
```

Comment:
I think this was done for making editable installs work for the projects using `setuptools` as build system. Seems like it is supported since [v64.0.0](https://setuptools.pypa.io/en/latest/history.html#v64-0-0). In any case, we are using `hatch` here and it already supports editable installs and I agree that we can safely remove it.